homeassistant/components/light/mysensors.py

```diff
@@ -109,118 +117,268 @@ def device_state_attributes(self):
+
+class MySensorsLightDimmer(MySensorsLight):
+    """Dimmer child class to MySensorsLight."""
+
+    def __init__(self, *args):
+        """Setup instance attributes."""
+        super().__init__(*args)
+
+    def turn_on(self, **kwargs):
+        """Turn the device on."""
+        super()._turn_on_light()
+        super()._turn_on_dimmer(**kwargs)
+        super()._turn_on()
+
+    def turn_off(self, **kwargs):
+        """Turn the device off."""
+        ret = super()._turn_off_dimmer()
+        ret = super()._turn_off_light(
+            value_type=ret[ATTR_VALUE_TYPE], value=ret[ATTR_VALUE])
+        super()._turn_off(
+            value_type=ret[ATTR_VALUE_TYPE], value=ret[ATTR_VALUE])
+
+    def update(self):
+        """Update the controller with the latest value from a sensor."""
+        super()._update()
+        super()._update_light()
+        super()._update_dimmer()
+
+
+class MySensorsLightRGB(MySensorsLight):
+    """RGB child class to MySensorsLight."""
+
+    def __init__(self, *args):
+        """Setup instance attributes."""
+        super().__init__(*args)
+
+    def turn_on(self, **kwargs):
+        """Turn the device on."""
+        super()._turn_on_light()
+        super()._turn_on_dimmer(**kwargs)
+        super()._turn_on_rgb_and_w('%02x%02x%02x', **kwargs)
+        super()._turn_on()
+
+    def turn_off(self, **kwargs):
+        """Turn the device off."""
+        ret = super()._turn_off_rgb_or_w()
+        ret = super()._turn_off_dimmer(
+            value_type=ret[ATTR_VALUE_TYPE], value=ret[ATTR_VALUE])
+        ret = super()._turn_off_light(
+            value_type=ret[ATTR_VALUE_TYPE], value=ret[ATTR_VALUE])
+        super()._turn_off(
+            value_type=ret[ATTR_VALUE_TYPE], value=ret[ATTR_VALUE])
+
+    def update(self):
+        """Update the controller with the latest value from a sensor."""
+        super()._update()
+        super()._update_light()
+        super()._update_dimmer()
+        super()._update_rgb_or_w()
+
+
+class MySensorsLightRGBW(MySensorsLight):
+    """RGBW child class to MySensorsLight."""
+
+    def __init__(self, *args):
+        """Setup instance attributes."""
+        super().__init__(*args)
+
+    def turn_on(self, **kwargs):
+        """Turn the device on."""
+        super()._turn_on_light()
+        super()._turn_on_dimmer(**kwargs)
+        super()._turn_on_rgb_and_w('%02x%02x%02x%02x', **kwargs)
+        super()._turn_on()
+
+    def turn_off(self, **kwargs):
+        """Turn the device off."""
+        ret = super()._turn_off_rgb_or_w()
+        ret = super()._turn_off_dimmer(
+            value_type=ret[ATTR_VALUE_TYPE], value=ret[ATTR_VALUE])
+        ret = super()._turn_off_light(
+            value_type=ret[ATTR_VALUE_TYPE], value=ret[ATTR_VALUE])
+        super()._turn_off(
+            value_type=ret[ATTR_VALUE_TYPE], value=ret[ATTR_VALUE])
+
+    def update(self):
+        """Update the controller with the latest value from a sensor."""
+        super()._update()
```

Comment:
You should only use `super()` if you're calling a method on the parent that you overwrote. Since you didn't overwrite this method, you can call `self._update()`